source/administration/plugins.rst

```diff
@@ -0,0 +1,63 @@
+Plugins (Beta)
+===============
+
+Mattermost supports plugins, which allow users to customize and extend the platform by adding specific features.
+
+There are three scenarios in which you would consider using a plugin:
+
+ - When you want to customize the Mattermost user interface
+ - When you want to extend Mattermost functionality to meet a specific, complex requirement
+ - When you want to build integrations that are managed by your Mattermost server
+
+There are two main types of plugins:
+
+ - **Client plugins**: Allows you to customize the Mattermost user interface, by overriding elements such as the profile popover, channel header or the sidebar. For a sample plugin, see `hovercardexample <https://github.com/jwilander/hovercardexample>`_.
+ - **Server plugins**: Makes it easier to integrate with third-party systems such as JIRA, GitLab or Jenkins. A sample plugin for a video and audio call system such as Zoom and Skype for Business is currently in progress.
+
+Later in upcoming releases, the `Enterprise Edition <https://about.mattermost.com/pricing>`_ will support server plugins that enable you to extend Mattermost functionality to meet a specific, complex requirement such as profiling performance metrics, and to implement highly customized compliance rules for `information barriers <http://www.17a-4.com/supervision-information-barriers/>`_.
+
+Set Up Guide
+--------------
+
+To enable plugins, follow these two simple steps.
+
+1) Go to **System Console > Plugins (Beta) > Configuration**. Here you can enable plugins and plugin uploads. If you do not plan to upload a plugin, set **Enable Plugin Uploads** to ``false`` to control which plugins are installed on your server. 
+2) Go to **System Console > Plugins (Beta) > Management** to manage your plugins, including:
+
+ - Activating or deactivating pre-packaged Mattermost plugins.
+ - Upload and install a plugin for your Mattermost server, or remove them.
+
+For more information on building your own plugin, see the `developer documentation <//xxx Joram, anything we can reference?>`_.
```

Comment:
Link needs to be updated. Do we have developer docs for this? Seems like a big part of the documentation that still needs to be done on how to actually create the plugins?

Reply:
Yeah, we have some early documentation for this. @jwilander Given the developers.mattermost.com won't be quite ready, maybe we should add the docs to docs.mattermost.com for now so there's a reference?

Reply:
I can copy some of the markdown we have for developers.mattermost.com over to here but a lot of it is generated and not super easy to migrate over

Reply:
@jwilander We can also consider linking to https://github.com/jwilander/hovercardexample as an example, since it has details on how the plugin was built.